### ISSUE TITLE
Training terminates while attempting to generate samples on Mac

### ISSUE THREAD

**Issue by kelphead** [bmaltais/kohya_ss#493]
**This is a bug report and a workaround for those in my situation**

My setup is on a Mac M2 air with the following config:

```
In which compute environment are you running?
This machine
      
Which type of machine are you using?         
No distributed training 
                                     
Do you want to run your training on CPU only (even if a GPU is available)? [yes/NO]:yes   
        
Do you wish to optimize your script with torch dynamo?[yes/NO]:no       
                                   
Do you want to use DeepSpeed? [yes/NO]: no                         
                                         
Do you wish to use FP16 or BF16 (mixed precision)?         
no            
```

My training was going fine until it tried to generate samples, at which point I got this error `AssertionError: Torch not compiled with CUDA enabled` and my training stopped. I was able to get it to work with the following workaround

**Workaround**:
comment out the following lines:
https://github.com/bmaltais/kohya_ss/blob/master/library/train_util.py#L2930
https://github.com/bmaltais/kohya_ss/blob/master/library/train_util.py#L2987
https://github.com/bmaltais/kohya_ss/blob/master/library/train_util.py#L3022
https://github.com/bmaltais/kohya_ss/blob/master/library/train_util.py#L3025

Basically, go to train_util.py and comment out everything that mentions cuda

**Comment by bmaltais** [comment]
Unfortunately this is code that Kohya create and modify. Every time there is an update you will need to modify it. Better option might be to report it to Kohya directly and see if he can help.

**Comment by jstayco** [comment]
When you commented out those cuda lines did it work as expected? You got the expected output? Or it just appeared to be working, but the output was no good?

**Comment by SemLSummeR** [comment]
> 

My Macbook is also m2 air. When training the rola model, an error was reported when it was almost over. Did you change the train_util.py file successfully train the Lora model? If so, can you tell me how to change it?

**Comment by ourcolour** [comment]
+1

**Comment by ourcolour** [comment]
+!
Torch not compiled with CUDA enabled

**Comment by savana10** [comment]
> I've tried this way on My Macbook Pro M2:
> 
> Mixed Precision: no Save Precision: float Optimizer: AdamW Advanced Configuraion: **UN-check** xformers

this worked for me
